### PR TITLE
[Enhancement] Refactor ObjectType and PrivilegeType from enum to class 

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/privilege/RolePrivilegeCollectionDeprecate.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.privilege;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class RolePrivilegeCollectionDeprecate extends PrivilegeCollectionDeprecate {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/privilege/RolePrivilegeCollectionDeprecate.java:21]: what is this meant to?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 734 / 931 (78.84%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|0|5|00.00%|[189, 229, 514, 1110, 1114]|
|:large_blue_circle:|com/starrocks/catalog/system/starrocks/RoleEdges.java|0|4|00.00%|[53, 60, 74, 81]|
|:large_blue_circle:|com/starrocks/catalog/system/starrocks/GrantsTo.java|56|170|32.94%|[138, 139, 140, 141, 142, 143, 144, 145, 148, 149, 150, 151, 153, 158, 159, 160, 161, 162, 164, 165, 166, 167, 170, 180, 181, 187, 198, 199, 200, 201, 202, 204, 205, 206, 207, 210, 215, 222, 223, 230, 241, 245, 246, 255, 256, 257, 259, 263, 264, 265, 266, 267, 268, 270, 271, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 285, 286, 287, 289, 295, 296, 297, 301, 302, 304, 305, 306, 307, 310, 313, 316, 317, 318, 319, 332, 333, 334, 336, 337, 338, 339, 341, 342, 343, 344, 346, 348, 349, 350, 351, 352, 353, 354, 355, 357, 358, 359, 361, 362, 363, 365, 366, 367, 369]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeCollection.java|3|6|50.00%|[249, 268, 269]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|1|2|50.00%|[1797]|
|:large_blue_circle:|com/starrocks/privilege/ObjectTypeDeprecate.java|42|67|62.69%|[41, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 70, 99]|
|:large_blue_circle:|com/starrocks/privilege/RolePrivilegeCollection.java|5|6|83.33%|[76]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|6|83.33%|[2124]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|124|140|88.57%|[252, 323, 324, 325, 329, 344, 353, 377, 396, 420, 421, 422, 424, 437, 457, 467]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeEntry.java|20|22|90.91%|[62, 64]|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|163|179|91.06%|[272, 1041, 1651, 1652, 1682, 1683, 1750, 1751, 1849, 1852, 1867, 1873, 1904, 1907, 1913, 1916]|
|:large_blue_circle:|com/starrocks/privilege/ObjectType.java|44|47|93.62%|[44, 52, 117]|
|:large_blue_circle:|com/starrocks/privilege/UserPrivilegeCollectionV2.java|16|17|94.12%|[59]|
|:large_blue_circle:|com/starrocks/privilege/DefaultAuthorizationProvider.java|16|17|94.12%|[159]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|42|44|95.45%|[42, 118]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeCollectionV2.java|137|139|98.56%|[51, 289]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ResourceGroupMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeActions.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/RolePrivilegeCollectionV2.java|53|53|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 733 / 928 (78.99%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/mysql/privilege/AuthUpgrader.java|0|5|00.00%|[189, 229, 514, 1110, 1114]|
|:large_blue_circle:|com/starrocks/catalog/system/starrocks/RoleEdges.java|0|4|00.00%|[53, 60, 74, 81]|
|:large_blue_circle:|com/starrocks/catalog/system/starrocks/GrantsTo.java|56|170|32.94%|[138, 139, 140, 141, 142, 143, 144, 145, 148, 149, 150, 151, 153, 158, 159, 160, 161, 162, 164, 165, 166, 167, 170, 180, 181, 187, 198, 199, 200, 201, 202, 204, 205, 206, 207, 210, 215, 222, 223, 230, 241, 245, 246, 255, 256, 257, 259, 263, 264, 265, 266, 267, 268, 270, 271, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 285, 286, 287, 289, 295, 296, 297, 301, 302, 304, 305, 306, 307, 310, 313, 316, 317, 318, 319, 332, 333, 334, 336, 337, 338, 339, 341, 342, 343, 344, 346, 348, 349, 350, 351, 352, 353, 354, 355, 357, 358, 359, 361, 362, 363, 365, 366, 367, 369]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeCollection.java|3|6|50.00%|[249, 268, 269]|
|:large_blue_circle:|com/starrocks/privilege/ObjectTypeDeprecate.java|42|67|62.69%|[41, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 70, 99]|
|:large_blue_circle:|com/starrocks/privilege/RolePrivilegeCollection.java|5|6|83.33%|[76]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|124|140|88.57%|[252, 323, 324, 325, 329, 344, 353, 377, 396, 420, 421, 422, 424, 437, 457, 467]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeEntry.java|20|22|90.91%|[62, 64]|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|163|179|91.06%|[272, 1041, 1651, 1652, 1682, 1683, 1750, 1751, 1849, 1852, 1867, 1873, 1904, 1907, 1913, 1916]|
|:large_blue_circle:|com/starrocks/privilege/ObjectType.java|44|47|93.62%|[44, 52, 117]|
|:large_blue_circle:|com/starrocks/privilege/UserPrivilegeCollectionV2.java|16|17|94.12%|[59]|
|:large_blue_circle:|com/starrocks/privilege/DefaultAuthorizationProvider.java|16|17|94.12%|[159]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeType.java|42|44|95.45%|[42, 118]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeCollectionV2.java|137|139|98.56%|[51, 289]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/authentication/AuthenticationMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ResourceGroupMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeActions.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/privilege/RolePrivilegeCollectionV2.java|53|53|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|3|3|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**